### PR TITLE
Update Contributing documents

### PR DESCRIPTION
**Pull request by ma11hew28**
- Add `brew uninstall foo`
- Add `--build-from-source` to `brew install foo`
- Add `--strict --online` to `brew audit foo`

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ First time contributing to Homebrew? Read our [Code of Conduct](https://github.c
 
 * `brew edit foo` and make edits
 * leave the [`bottle`](http://www.rubydoc.info/github/Homebrew/brew/master/Formula#bottle-class_method) as-is
-* `brew install foo`, `brew test foo`, and `brew audit foo`
+* `brew uninstall foo`, `brew install --build-from-source foo`, `brew test foo`, and `brew audit --strict --online foo`
```
Comment (MikeMcQuaid) [inline .github/CONTRIBUTING.md]: `brew reinstall --build-from-source foo`. We don't actually need `--strict --online` for making fixes, just `audit`.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 - [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
 - [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
-- [ ] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
+- [ ] Have you built your formula locally prior to submission with `brew install --build-from-source <formula>` (where `<formula>` is the name of the formula you're submitting)?
```
Comment (MikeMcQuaid) [inline .github/PULL_REQUEST_TEMPLATE.md]: `brew reinstall --build-from-source`

**Comment by MikeMcQuaid** [comment]
Thanks for these!

**Comment by mattdipasquale** [comment]
Sure. Thanks for the feedback! So, should I make those changes you suggested and resubmit the pull request, or should I let you guys do that?

**Comment by MikeMcQuaid** [comment]
If you could make those changes in this PR and push them either as a new commit or (ideally) by squashing into the original commit that'd be great. Thanks!

**Comment by ilovezfs** [comment]
I think I might prefer recommending "uninstall" + "install --build-from-source" instead of "reinstall --build-from-source" since we really want these tested with the default option set, not the user's preexisting option set, unless the change is specifically in regard to one of the non-default options, with possibly a note that existing options won't be preserved, so you may need to re-customize your local install following testing.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (MikeMcQuaid) [inline .github/CONTRIBUTING.md]: Good point from @ilovezfs: let's go for `brew uninstall --force foo`, `brew install --build-from-source foo`

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (MikeMcQuaid) [inline .github/PULL_REQUEST_TEMPLATE.md]: Good point from @ilovezfs: let's go for `brew uninstall --force foo`, `brew install --build-from-source foo`

**Comment by dunn** [comment]
Mind rebasing this against master?

**Comment by MikeMcQuaid** [comment]
@mattdipasquale Gentle ping on this when you get a chance.

**Comment by mattdipasquale** [comment]
@dunn @MikeMcQuaid sorry guys. While making the updates, I got carried away and never finished and forgot about it. I can't find the latest changes I made. I don't think I'll be able to get to this until a few days. Feel free to take over if it's urgent.

**Comment by MikeMcQuaid** [comment]
@mattdipasquale We can't take over your PR, unfortunately, so we'll wait a little longer. Thanks!

**Comment by mattdipasquale** [comment]
Instead of "squashing the original commit" @MikeMcQuaid or "rebasing against master" @dunn, I just merged master into this branch and made another commit with the fixes because I wanted to keep (for reference) the already published history: this conversation, your inline comments, and the progression of my changes. I hope that's OK.

**Comment by zmwangx** [comment]
I propose `brew test` be added to the checklist. I hate long issue/PR templates as much as the next guy, but it's kind of strange that audit is in and test is out.

**Comment by MikeMcQuaid** [comment]
@zmwangx I'd rather we keep it as-is for now but good idea to consider in future 👍

**Comment by zmwangx** [comment]
Is this abandoned?

**Comment by MikeMcQuaid** [comment]
@zmwangx Even if so: we should recreate ourselves, I think.

**Comment by mattdipasquale** [comment]
Why wasn't this branch merged into master?

**Comment by MikeMcQuaid** [comment]
@mattdipasquale There's merge conflicts that need addressed.

**Comment by MikeMcQuaid** [comment]
@mattdipasquale I'll 🚢 this after they are.

**Comment by MikeMcQuaid** [comment]
Sorry for the delay on this! Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by mattdipasquale** [comment]
That's OK! You're welcome. Glad I was able to help. Thank you. You rock too! :-)

**Pull request closed without merging**